lib/msal-node/src/cache/Storage.ts

```diff
@@ -49,80 +39,118 @@ export class Storage extends CacheManager {
+            } else if (cache[key] instanceof IdTokenEntity) {
+                inMemoryCache.idTokens[key] = cache[key] as IdTokenEntity;
+            } else if (cache[key] instanceof AccessTokenEntity) {
+                inMemoryCache.accessTokens[key] = cache[key] as AccessTokenEntity;
+            } else if (cache[key] instanceof RefreshTokenEntity) {
+                inMemoryCache.refreshTokens[key] = cache[key] as RefreshTokenEntity;
+            } else if (cache[key] instanceof AppMetadataEntity) {
+                inMemoryCache.appMetadata[key] = cache[key] as AppMetadataEntity;
+            } else {
+                continue;
+            }
+        }
+
+        return inMemoryCache;
+    }
+
+    /**
+     * converts inMemoryCache to CacheKVStore
+     * @param inMemoryCache
+     */
+    inMemoryCacheToCache(inMemoryCache: InMemoryCache): CacheKVStore {
+        // convert in memory cache to a flat Key-Value map
+        let cache = this.getCache();
+
+        cache = {
+            ...inMemoryCache.accounts,
+            ...inMemoryCache.idTokens,
+            ...inMemoryCache.accessTokens,
+            ...inMemoryCache.refreshTokens,
+            ...inMemoryCache.appMetadata
+        }
+        return cache;
+    }
+
     /**
      * gets the current in memory cache for the client
      */
-    getCache(): object {
+    getInMemoryCache(): object {
         this.logger.verbose("Getting in-memory cache");
-        return this.inMemoryCache;
+
+        // convert the cache key value store to inMemoryCache
+        const inMemoryCache = this.cacheToInMemoryCache(this.getCache());
+        return inMemoryCache;
     }
 
     /**
      * sets the current in memory cache for the client
      * @param inMemoryCache
      */
-    setCache(inMemoryCache: InMemoryCache) {
+    setInMemoryCache(inMemoryCache: InMemoryCache): void{
         this.logger.verbose("Setting in-memory cache");
-        this.inMemoryCache = inMemoryCache;
+
+        // convert and append the inMemoryCache to cacheKVStore
+        const cache = this.inMemoryCacheToCache(inMemoryCache);
+        this.setCache(cache);
+
+        this.emitChange();
+    }
+
+    /**
+     * get the current cache key-value store
+     */
+    getCache(): CacheKVStore {
+        this.logger.verbose("Getting cache key-value store");
+        return this.cache;
+    }
+
+    /**
+     * sets the current cache (key value store)
+     * @param cacheMap
+     */
+    setCache(cache: CacheKVStore): void {
+        this.logger.verbose("Setting cache key value store");
+        this.cache = cache;
+
+        // mark change in cache
         this.emitChange();
     }
 
     /**
-     * Set Item in memory
+     *
      * @param key
      * @param value
      * @param type
-     * @param inMemory
      */
-    setItem(
-        key: string,
-        value: string | object,
-        type?: string
-    ): void {
+    setItem(key: string, value: ValidCacheType, type?: string) {
```

Comment:
If `type` is not provided, will this store and return a string in `getItem`?

Reply:
The idea is to get rid of type altogether. I kept it until tests make it obsolete.